hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/embedded/EmbeddedTimelineService.java

```diff
@@ -124,10 +124,8 @@ public FileSystemViewManager getViewManager() {
     return viewManager;
   }
 
-  public boolean canReuseFor(String basePath) {
-    return this.server != null
-        && this.viewManager != null
-        && this.basePath.equals(basePath);
+  public boolean canReuse() {
+    return this.server != null && this.viewManager != null;
   }
```

Comment:
Why we reuse the same timeline service for different path here ?

Reply:
We don't
`canReuse()` is only used in one place -- `createEmbeddedTimelineService()`
I added `Map<String, EmbeddedTimelineService> PATH_TO_SERVER` to support multiple base paths
We can double check basePath in `canReuse()` but I think it is redundant

Reply:
Did you know that we currently used the timeline service started by the driver ? When the driver starts, it tries to start a `TimelineService` first, and all the workers then access this service, the service is per-job singleton.

Reply:
> started by the driver

you mean `StreamWriteOperatorCoordinator`?

> all the workers then access this service

as I can see it is true only for workers with the same `basePath`

Yes I understand that service is per-job singleton. In some way it is preserved by this PR: we have singleton for workers based on the same `basePath`

Reply:
Example:
```java
tEnv.executeSql("create table t1 (...) with ('path'='/tmp/t1', ...)")
tEnv.executeSql("create table t2 (...) with ('path'='/tmp/t2', ...)")
TableResult r1 = tEnv.executeSql("insert into t1 ...")
TableResult r2 = tEnv.executeSql("insert into t1 ...")
r1.await()                  // at this moment PATH_TO_SERVER = {/tmp/t1: timelineService1, /tmp/t2: timelineService2}
r2.await()
```

Reply:
All the workers would try to access singleton remote timeline service from config options lies in ${base_path}/.aux/view_storage_conf.properties

Reply:
Yes I argee with you. This PR does not change it. I'm talking about we need two singltons of timeline service when we have two different ${base_path}. It is possible in example above
`/tmp/t1/.aux/view_storage_conf.properties`
`/tmp/t2/.aux/view_storage_conf.properties`

Reply:
Well I can check what happens if both `/tmp/t1/.aux/view_storage_conf.properties`, `/tmp/t2/.aux/view_storage_conf.properties` contain the same timeline service but this logic it is missing on the lates master. Currenlty, we start new timeline service and reassign `Option<EmbeddedTimelineService> TIMELINE_SERVER` when new `basePath` is passed

Reply:
I've checked the same timeservice for different `basePath`. It works on my test. Do you think we need to reuse the same timeline service for different path?